bitsail-connectors/connector-ftp/src/main/java/com/bytedance/bitsail/connector/ftp/source/reader/DeserializationSchemaFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.connector.ftp.source.reader;
+
+import com.bytedance.bitsail.base.connector.reader.v1.SourceReader;
+import com.bytedance.bitsail.base.format.DeserializationSchema;
+import com.bytedance.bitsail.common.configuration.BitSailConfiguration;
+import com.bytedance.bitsail.component.format.csv.CsvDeserializationSchema;
+import com.bytedance.bitsail.component.format.json.JsonDeserializationSchema;
+import com.bytedance.bitsail.connector.ftp.core.config.FtpConfig;
+
+public class DeserializationSchemaFactory {
+  public static DeserializationSchema createDeserializationSchema(BitSailConfiguration jobConf, FtpConfig ftpConfig, SourceReader.Context context) {
+    DeserializationSchema deserializationSchema;
+    if (ftpConfig.getContentType() == FtpConfig.ContentType.CSV) {
+      deserializationSchema = new CsvDeserializationSchema(
+          jobConf,
+          context.getTypeInfos(),
+          context.getFieldNames());;
+    } else {
```

Comment:
we could throw an supported exception if other than csv and json

Reply:
I will add it in next revision.